litellm/llms/replicate.py

```diff
@@ -201,38 +201,35 @@ def completion(
         if k not in optional_params: # completion(top_k=3) > replicate_config(top_k=3) <- allows for dynamic variables to be passed in
             optional_params[k] = v
 
-    if "meta/llama-2-13b-chat" in model: 
-        system_prompt = ""
-        prompt = "" 
-        for message in messages: 
-            if message["role"] == "system":
-                system_prompt = message["content"]
-            else: 
-                prompt += message["content"]
-        input_data = {
-            "system_prompt": system_prompt,
-            "prompt": prompt,
-            **optional_params
-        }
-    else:
-        if model in custom_prompt_dict:
-            # check if the model has a registered custom prompt
-            model_prompt_details = custom_prompt_dict[model]
-            prompt = custom_prompt(
-                    role_dict=model_prompt_details.get("roles", {}), 
-                    initial_prompt_value=model_prompt_details.get("initial_prompt_value", ""),  
-                    final_prompt_value=model_prompt_details.get("final_prompt_value", ""), 
-                    bos_token=model_prompt_details.get("bos_token", ""),
-                    eos_token=model_prompt_details.get("eos_token", ""),
-                    messages=messages,
-                )
+    # extract system prompt
+    system_prompt = ""
+    prompt_messages = []
+    for message in messages:
+        if message["role"] == "system":
+            system_prompt = message["content"]
         else:
-            prompt = prompt_factory(model=model, messages=messages)
+            prompt_messages += [message]
+            
+    # apply template if provided 
+    if model in custom_prompt_dict:
+        # check if the model has a registered custom prompt
+        model_prompt_details = custom_prompt_dict[model]
+        prompt = custom_prompt(
+                role_dict=model_prompt_details.get("roles", {}), 
+                initial_prompt_value=model_prompt_details.get("initial_prompt_value", ""),  
+                final_prompt_value=model_prompt_details.get("final_prompt_value", ""), 
+                bos_token=model_prompt_details.get("bos_token", ""),
+                eos_token=model_prompt_details.get("eos_token", ""),
+                messages=prompt_messages,
+            )
+    else:
+        prompt = prompt_factory(model=model, messages=prompt_messages)
 
-        input_data = {
-            "prompt": prompt,
-            **optional_params
-        }
+    input_data = {
```

Comment:
@wendlerc replicate only has a system prompt for some models (e.g. llama), not all. see - https://replicate.com/01-ai/yi-34b-chat